pkg/stream/multi_producer_test.go

```diff
@@ -0,0 +1,198 @@
+package stream_test
+
+import (
+	"context"
+	"errors"
+	"github.com/applike/gosoline/pkg/stream"
+	"github.com/applike/gosoline/pkg/stream/mocks"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func newMessage() interface{} {
+	return &stream.Message{
+		Attributes: map[string]interface{}{
+			stream.AttributeEncoding: stream.EncodingJson,
+		},
+		Body: `{"id":3,"name":"foobar"}`,
+	}
+}
+
+func newAttributeSetsParams() []map[string]interface{} {
+	return []map[string]interface{}{
+		{
+			"test": "foo",
+		},
+		{
+			"test": 100,
+		},
+	}
+}
+
+func newParams(context context.Context, model interface{}, sets []map[string]interface{}) []interface{} {
+	params := []interface{}{context, model}
+	for _, set := range sets {
+		params = append(params, set)
+	}
+	return params
+}
+
+func TestMultiProducerWriteOne(t *testing.T) {
+	type mockConf struct {
+		err        error
+		expectCall bool
+	}
+
+	testErr := errors.New("testErr")
+
+	tt := []struct {
```

Comment:
please don't create inline structs

Reply:
I'll restructure the tests a bit to get rid of the duplication.

This is a fairly standard pattern for table driven tests in Go though, since the structs are generally local to the specific test. So I don't really see the harm in using it.

Reply:
but you using it in both tests, so why not remove code duplication here?

Reply:
Those are seperate issues right, not using inline structs in general vs removing duplicate code. 
I'll push a new solution in a bit.

Reply:
your last push didn't contain a fix here?

Reply:
The testing table is fully local to `TestMultiProducerTestCallingOfChildren` at the moment. So there's no need to define a type for it.